internal/resources/network_peer.go

```diff
@@ -0,0 +1,418 @@
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error reading private endpoint status",
+			"Error reading private endpoint status, unexpected error: "+errString,
+		)
+
+		return
+	}
+
+	diags = resp.State.Set(ctx, &refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update there is no update API so returns an error.
+func (n *NetworkPeer) Update(_ context.Context, _ resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError(
+		"No update API for network peering",
+		"There doesn't exist an update API for network peering",
+	)
+}
+
+// Delete deletes a network peer of the CSP.
+func (n *NetworkPeer) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state providerschema.NetworkPeer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting network peer ",
+			"Could not delete network peer id:"+state.Id.String()+"due to validation error: "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		peerId         = IDs[providerschema.Id]
+	)
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/networkPeers/%s",
+		n.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		peerId,
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodDelete, SuccessStatus: http.StatusNoContent}
+	_, err = n.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		n.Token,
+		nil,
+	)
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error Deleting network peer",
+			"Could not delete network peer id "+state.Id.String()+": "+errString,
+		)
+		return
+	}
+}
+
+func (n *NetworkPeer) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *providerschema.Data, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	n.Data = data
+}
+
+func (n *NetworkPeer) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (n *NetworkPeer) validateCreateNetworkPeer(plan providerschema.NetworkPeer) error {
+	if plan.OrganizationId.IsNull() {
```

Comment:
move null checks to validate function

Reply:
I think this is fine, the null checks will be validated here itself- would prefer to keep this separate validate func for Create (resources-`NetworkPeer` struct). The other `Validate` function is for (datasources-`NetworkPeers` struct)